File: Queue.cpp
#include <iostream>
#include "Queue.h"

Queue :: Queue(){
    first = NULL;
    last = NULL;
}
void Queue :: enqueue(int date){
    Node *node = new Node();
	node->setLoad(date);
	node->setNext(NULL);
	if(first==NULL){
		first = node;
		last = node;	      
	}else{
		last->setNext(node);
		last = node;
	}
}

int Queue :: dequeue(){
    if(isEmpty()){
		return -1;
	}else{
		Node *temp = first;
		int ourValue = first->getLoad();
		first = first->getNext();
		free(temp);
		return ourValue;
	}
}

void Queue :: clear(){
do{
    dequeue();
}while(first!=NULL);
}

void Queue :: peek(){
    if(isEmpty())
        std::cout<<"It's empty";
    else
        std::cout<<last->getLoad();
}

bool Queue :: isEmpty(){
    return (first==NULL) ? true : false;
}

void Queue :: toString(){
	std:: cout<<"[";
		Node *temp = first;
		while(temp!=NULL){
			if(temp->getNext()==NULL){
				std::cout<<temp->getLoad();
			}else{
				std::cout<<temp->getLoad();
				std::cout<<",";
			}
			temp = temp->getNext();
		}
	std::cout<<"]";
}

File: Queue.java
public class Queue{

        private Node first = null;
        private Node last = null;

    public void enqueue(int date){
        Node node = new Node(date);
        node.setLoad(date);
        node.setNext(null);
        if(first==null){
            first = node;
            last = node;	      
        }else{
            last.setNext(node);
            last = node;
        }
    }
    
    public int dequeue(){
        if(isEmpty()){
            return -1;
        }else{
            Node temp = first;
            int ourValue = first.getLoad();
            first = first.getNext();
            return ourValue;
        }
    }
    
    public void clear(){
    do{
        dequeue();
    }while(first!=null);
    }
    
    public void peek(){
        if(isEmpty())
            System.out.println("It's empty");
        else
        System.out.println(last.getLoad());
    }
    
    public boolean isEmpty(){
        return (first==null) ? true : false;
    }
    
    public String toString(){
        String brackets = "[";
        String brackets2 = "]";
        if(isEmpty()==true)
            return brackets+"NULL"+brackets2;
        else{
            Node current = first;
            while(current.getNext() !=null){
                if(current!=last)
                    brackets = brackets + current.getLoad()+",";
                else
                    brackets = brackets + current.getLoad();
            current = current.getNext();
            }
            brackets = brackets+current.getLoad()+brackets2;
        }
        return brackets;
    }
}
